src/Types.ts

```diff
@@ -9,10 +9,27 @@ type DataPath = "mounted" | "default" | "upload";
 // Fills input for the "Data:" dropdown in the HeaderForm
 type DataType = "built-in" | "file-upload" | "mounted files" | "examples";
 
+// Possible filestypes taken from the request
+// Files like GBZ contains graph and maybe haplotype
+type filetype = "graph" | "haplotype" | "read" | "bed" | "graph-haplotype";
```

Comment:
If we're using the GBZ file with graph type and with haplotype type to indicate whether we want its haplotypes or not, we don't need the combined type anymore.
```suggestion
// Files like GBZ contains graph and maybe haplotype and so can be either
type filetype = "graph" | "haplotype" | "read" | "bed";
```